src/components/Proposal.jsx

```diff
@@ -89,14 +94,60 @@ const cmp = (a, b) => {
 const sortedEntries = entries =>
   Object.entries(entries).sort(([kwa], [kwb]) => cmp(kwa, kwb));
 
-const Proposal = ({ offer, purses }) => {
+const Proposal = ({ offer, purses, swingsetParams, beansOwing }) => {
+  const [anchorEl, setAnchorEl] = React.useState(null);
+
+  const handleClick = event => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const isOpen = Boolean(anchorEl);
+
   const {
     proposalForDisplay,
     proposalTemplate,
     invitationDetails: { fee, feePursePetname, expiry } = {},
     error,
+    spendAction,
   } = offer;
 
+  const feeUnit = swingsetParams?.beansPerUnit?.find(
+    ({ key }) => key === 'feeUnit',
+  )?.beans;
+  const minFeeDebit = swingsetParams?.beansPerUnit?.find(
+    ({ key }) => key === 'minFeeDebit',
+  )?.beans;
+  const messageCost = swingsetParams?.beansPerUnit?.find(
+    ({ key }) => key === 'message',
+  )?.beans;
+  const messageByteCost = swingsetParams?.beansPerUnit?.find(
+    ({ key }) => key === 'messageByte',
+  )?.beans;
+  const inboundTxCost = swingsetParams?.beansPerUnit?.find(
+    ({ key }) => key === 'inboundTx',
+  )?.beans;
+  const feeThreshold =
+    feeUnit && minFeeDebit && Number(minFeeDebit) / Number(feeUnit);
+  const accumulatedFees =
+    beansOwing &&
+    feeUnit &&
+    minFeeDebit &&
+    (beansOwing % Number(minFeeDebit)) / Number(feeUnit);
+  const feeDelta =
+    messageCost &&
+    inboundTxCost &&
+    messageByteCost &&
+    feeUnit &&
+    spendAction &&
+    (Number(messageCost) +
+      Number(inboundTxCost) +
+      Number(messageByteCost) * spendAction.length) /
+      Number(feeUnit);
```

Comment:
suggestion:

 - add a comment to say that this calculation reproduces the logic in https://github.com/Agoric/agoric-sdk/blob/master/golang/cosmos/x/swingset/types/msgs.go
 - factor the calculation out as its own test
 - add unit tests using the data I provided in https://github.com/Agoric/wallet-app/pull/35#issuecomment-1304405274

Reply:
Pulled this out into its own component and added tests. The tests are at the component level since it's better to test the functionality as close as possible to how it will look in production.